### PR TITLE
Windows dockerfile

### PR DESCRIPTION
**Pull request by ellimac54**
Windows dockerfile for windows server 2019. All tools and full environment is installed in order to build and pass tests

### CODE REVIEW

**Comment by simon-mo** [comment]
Few comments to start:
- can we install miniconda, jdk, bazel, and git using package manager? so we can avoid Invoke-WebRequest if we can
- do we need all the *.cab files?
- can we remove dotnet and internet explorer related script? I don't think they are needed to build and test ray
- do we need all the *Helper.ps1 scripts?

**Comment by pcmoritz** [comment]
Please make sure to remove the binary `.cab` files from the PR before getting this merged (like `Microsoft-Windows-InternetExplorer-Optional-Package~31bf3856ad364e35~amd64~en-US~.cab` and there are a few more).

File: .buildkite/windows/install/post-generation/InternetExplorerConfiguration.ps1 [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+$AdminKey = "HKLM:\SOFTWARE\Microsoft\Active Setup\Installed Components\{A509B1A7-37EF-4b3f-8CFC-4F3A74704073}"
+$UserKey = "HKLM:\SOFTWARE\Microsoft\Active Setup\Installed Components\{A509B1A8-37EF-4b3f-8CFC-4F3A74704073}"
+Set-ItemProperty -Path $AdminKey -Name "IsInstalled" -Value 0 -Force
+Set-ItemProperty -Path $UserKey -Name "IsInstalled" -Value 0 -Force
+# restart Explorer process
+$ieProcess = Get-Process -Name Explorer -ErrorAction Ignore
+if ($ieProcess) {
+    Stop-Process -Name Explorer -Force -ErrorAction Ignore
+}
+Write-Host "IE Enhanced Security Configuration (ESC) has been disabled."
```
Comment (pcmoritz) [inline .buildkite/windows/install/post-generation/InternetExplorerConfiguration.ps1:15]: Why do we need this? Ray shouldn't be using the Internet Explorer, right?

File: .buildkite/windows/install/post-generation/RustJunction.ps1 [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Create Rust junction points to cargo and rustup folder
```
Comment (pcmoritz) [inline .buildkite/windows/install/post-generation/RustJunction.ps1:1]: Do we need Rust?

**Comment by simon-mo** [comment]
Closing in favor of #17474

**Pull request closed without merging**